e2e-polybft/e2e/nexus_bridge_test.go

```diff
@@ -1963,7 +1984,116 @@ func TestE2E_ApexBridgeWithNexus_BatchFailed(t *testing.T) {
 			require.False(t, timeout[i])
 		}
 
-		err = apex.WaitForExactAmount(ctx, user, cardanofw.ChainIDNexus, ethExpectedBalance, 100, time.Second*10)
+		err = apex.WaitForExactAmount(ctx, user, cardanofw.ChainIDNexus, ethExpectedBalance, 3, time.Second*10)
+		require.NoError(t, err)
+	})
+}
+
+func TestE2E_NexusFundAmount(t *testing.T) {
+	if cardanofw.ShouldSkipE2RRedundantTests() {
+		t.Skip()
+	}
+
+	const (
+		apiKey          = "test_api_key"
+		userCnt         = 15
+		fundAmountPrime = 100_000_000
+		fundAmountNexus = 100
+	)
+
+	ctx, cncl := context.WithCancel(context.Background())
+	defer cncl()
+
+	primeConfig := cardanofw.NewPrimeChainConfig()
+	primeConfig.FundAmount = 1_000_000
+
+	nexusConfig := cardanofw.NewNexusChainConfig(true)
+	nexusConfig.FundAmount = big.NewInt(1)
+
+	apex := cardanofw.SetupAndRunApexBridge(
+		t, ctx,
+		cardanofw.WithPrimeConfig(primeConfig),
+		cardanofw.WithNexusConfig(nexusConfig),
+		cardanofw.WithAPIKey(apiKey),
+		cardanofw.WithVectorEnabled(false),
+		cardanofw.WithNexusEnabled(true),
+		cardanofw.WithUserCnt(userCnt),
+	)
+
+	defer require.True(t, apex.ApexBridgeProcessesRunning())
+
+	user := apex.Users[userCnt-1]
+
+	fmt.Println("prime user addr: ", user.PrimeAddress)
+	fmt.Println("nexus user addr: ", user.NexusAddress)
+	fmt.Println("prime multisig addr: ", apex.PrimeInfo.MultisigAddr)
+	fmt.Println("prime fee addr: ", apex.PrimeInfo.FeeAddr)
+	fmt.Println("nexus gateway addr ", apex.NexusInfo.GatewayAddress)
+
+	t.Run("From nexus to prime - not enough funds", func(t *testing.T) {
```

Comment:
The code for these two tests is nearly identical; the only differences are the source chain, destination chain, and amounts. I would create two test cases and iterate over them directly, without using t.Run.